pkg/arvo/sys/zuse.hoon

```diff
@@ -2357,6 +2357,143 @@
   ++  blake
     ~%  %blake  ..part  ~
     |%
+    ++  blake3
+      |_  key=byts
+      ::
+      ++  cv     ?:(=(0 wid.key) iv dat.key)
+      ++  flags  ?:(=(0 wid.key) 0b0 f-keyedhash)
+      ::
+      +$  output
+        $:  cv=@ux
+            block=@ux
+            counter=@ud
+            blocklen=@ud
+            flags=@ub
+        ==
+      ::
+      ++  hash
+        |=  [msg=byts out=@ud]
+        ^-  @ux
+        %+  xof  out
+        %-  root-output
+        (turn (split-byts 13 msg) chunk-output)
+      ::
+      ++  xof
+        |=  [out=@ud o=output]
+        ^-  @
+        %^  rev  3  out
+        %+  rep  9
+        %+  turn  (gulf 0 (div out 64))
+        |=(i=@ (compress o(counter i)))
+      ::
+      ++  root-output
+        |=  outputs=(list output)
+        ^-  output
+        %+  set-flag  f-root
+        |-
+        =/  mid  (div (bex (xeb (dec (lent outputs)))) 2)
+        =+  [l=(scag mid outputs) r=(slag mid outputs)]
+        ?>  ?=(^ outputs)
+        ?~  t.outputs  i.outputs
+        (parent-output $(outputs l) $(outputs r))
+      ::
+      ++  parent-output
+        |=  [l=output r=output]
+        ^-  output
+        %+  set-flag  f-parent
+        [cv (rep 8 ~[(compress l) (compress r)]) 0 64 flags]
+      ::
+      ++  chunk-output
+        |=  [counter=@ chunk=byts]
+        ^-  output
+        %+  set-flag  f-chunkend
+        %+  roll  (split-byts 9 chunk)
+        |=  [[i=@ byts] prev=output]
+        ?:  =(0 i)  [cv dat counter wid (con flags f-chunkstart)]
+        [(rep 8 ~[(compress prev)]) dat counter wid flags]
+      ::
+      ++  compress
```

Comment:
`+compress` (and the other arms below) shouldn't be part of the public interface. What's the right way to hide them? Put them in a separate core and compose with `=>`?

Reply:
That's right, and `^?` the outer core so it can't be read thru.

Reply:
cool. `^?` breaks the door, so I had to change the keyed API, but it seems to work now.

Reply:
Oh, that's my bad. With a sample you want `^|`, not `^?` (%iron vs %lead variance).